src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/CreateClaimCallbackHandlerTest.java

```diff
@@ -141,6 +161,80 @@ void shouldReturnNoError_whenSoleTraderDateOfBirthIsInThePast() {
         }
     }
 
+    @Nested
+    class MidEventFeeCallback {
+
+        private final String pageId = "fee";
+        private final Fee feeData = Fee.builder()
+            .version("1")
+            .code("CODE")
+            .calculatedAmountInPence(BigDecimal.valueOf(100))
+            .build();
+        private final Organisation organisation = Organisation.builder()
+            .paymentAccount(List.of("12345", "98765"))
+            .build();
+
+        @BeforeEach
+        void setup() {
+            given(feesService.getFeeDataByClaimValue(any())).willReturn(feeData);
+        }
+
+        @Test
+        void shouldCalculateClaimFeeAndAddPbaNumbers_whenCalledAndOrgExistsInPrd() {
+            given(organisationService.findOrganisation(any())).willReturn(Optional.of(organisation));
+
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDraft().build();
+            CallbackParams params = callbackParamsOf(caseData, MID, pageId);
+
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getData())
+                .extracting("claimFee")
+                .extracting("calculatedAmountInPence", "code", "version")
+                .containsExactly(
+                    String.valueOf(feeData.getCalculatedAmountInPence()),
+                    feeData.getCode(),
+                    feeData.getVersion()
+                );
+
+            DynamicList dynamicList = getDynamicList(response);
+
+            List<String> actualPbas = dynamicList.getListItems().stream()
+                .map(DynamicListElement::getLabel)
+                .collect(Collectors.toList());
+
+            assertThat(actualPbas).containsOnly("12345", "98765");
+            assertThat(dynamicList.getValue()).isEqualTo(DynamicListElement.EMPTY);
+        }
+
+        @Test
+        void shouldCalculateClaimFee_whenCalledAndOrgDoesNotExistInPrd() {
+            given(organisationService.findOrganisation(any())).willReturn(Optional.empty());
+
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDraft().build();
+            CallbackParams params = callbackParamsOf(caseData, MID, pageId);
+
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getData())
+                .extracting("claimFee")
+                .extracting("calculatedAmountInPence", "code", "version")
+                .containsExactly(
+                    String.valueOf(feeData.getCalculatedAmountInPence()),
+                    feeData.getCode(),
+                    feeData.getVersion()
+                );
+
+            assertThat(getDynamicList(response))
+                .isEqualTo(DynamicList.builder().value(DynamicListElement.EMPTY).build());
+        }
+
+        private DynamicList getDynamicList(AboutToStartOrSubmitCallbackResponse response) {
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
this could be moved to field, so we don't create new instance on every method call